src/components/molecules/SectionRenderer/SectionRenderer.tsx

```diff
@@ -15,268 +13,115 @@ import * as S from './styled';
+    // disabled the below eslint rule because it was causing too many re-renders
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [rowIndexToScroll, virtualItems, lastScrollIndex]);
 
-  if (!sectionInstance?.isInitialized && sectionBlueprint.customization?.beforeInitializationText) {
+  if (!rootSectionInstance?.isInitialized && sectionBlueprint.customization?.beforeInitializationText) {
     return (
-      <S.BeforeInitializationContainer level={level}>
+      <S.BeforeInitializationContainer>
         <p>{sectionBlueprint.customization.beforeInitializationText}</p>
       </S.BeforeInitializationContainer>
     );
   }
 
-  if (!sectionInstance?.isVisible) {
+  if (!rootSectionInstance?.isVisible) {
     return null;
   }
 
-  if (sectionInstance?.isLoading) {
+  if (rootSectionInstance?.isLoading) {
     return <S.Skeleton />;
   }
 
-  if (sectionInstance?.isEmpty) {
+  if (rootSectionInstance?.isEmpty) {
     if (EmptyDisplay && EmptyDisplay.Component) {
       return (
-        <S.EmptyDisplayContainer level={level}>
-          <EmptyDisplay.Component sectionInstance={sectionInstance} />
+        <S.EmptyDisplayContainer>
+          <EmptyDisplay.Component sectionInstance={rootSectionInstance} />
         </S.EmptyDisplayContainer>
       );
     }
     return (
-      <S.EmptyDisplayContainer level={level}>
-        <h1>{sectionInstance.name}</h1>
+      <S.EmptyDisplayContainer>
+        <h1>{rootSectionInstance.name}</h1>
         <p>Section is empty.</p>
       </S.EmptyDisplayContainer>
     );
   }
 
   return (
-    <>
-      <SectionHeader
-        name={sectionInstance.name}
-        sectionInstance={sectionInstance}
-        sectionBlueprint={sectionBlueprint}
-        isCollapsed={isCollapsed}
-        isLastSection={isLastSection}
-        level={level}
-        expandSection={expandSection}
-        collapseSection={collapseSection}
-        indentation={sectionIndentation || 0}
-      />
-      {sectionInstance &&
-        sectionInstance.isVisible &&
-        !isCollapsed &&
-        itemBlueprint &&
-        sectionInstance.groups.length === 0 &&
-        sectionInstance.visibleItemIds.map(itemId => (
-          <ItemRenderer
-            key={itemId}
-            itemId={itemId}
-            blueprint={itemBlueprint}
-            level={level + 1}
-            isLastItem={isLastVisibleItemId(itemId)}
-            isSectionCheckable={Boolean(sectionInstance.checkable)}
-            sectionContainerElementId={sectionBlueprint.containerElementId}
-            options={itemRendererOptions}
-            indentation={sectionIndentation || 0}
-          />
-        ))}
-      {sectionInstance?.isVisible &&
-        !isCollapsed &&
-        itemBlueprint &&
-        groupInstanceById &&
-        sectionInstance.visibleGroupIds.map(groupId => {
-          const group = groupInstanceById[groupId];
+    <div style={{height: `${height}px`, overflow: 'auto'}} ref={parentRef}>
```

Comment:
Can we have this as a styled component?

Reply:
I'm not sure, I had problems with the virtualization when I tried making those as styled components. I'll probably just add comments reflecting that.